test/UriTest.php

```diff
@@ -159,22 +158,23 @@ public function testWithPortReturnsNewInstanceWithProvidedPort($port)
     public function testWithPortReturnsSameInstanceWithProvidedPortIsSameAsBefore()
     {
         $uri = new Uri('https://user:pass@local.example.com:3001/foo?bar=baz#quz');
-        $new = $uri->withPort('3001');
+        $new = $uri->withPort(3001);
         $this->assertSame($uri, $new);
-        $this->assertEquals('3001', $new->getPort());
+        $this->assertEquals(3001, $new->getPort());
```

Comment:
Internally, since we cast to an integer already, the above line should become:

```php
$this->assertSame(3001, $new->getPort());
```

That change should be made regardless of accepting the other portions of this patch.